crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -156,14 +156,17 @@ impl Extension for QueryLimitsChecker {
             depth: 0,
             num_nodes: 0,
         };
+        let mut max_depth_seen = 0;
```

Comment:
i see, so previously we were basically summing the max depth across all queries sent

still, it's possible for me to have a collection of queries, one that is invalid, but not intended to be executed. is this the tradeoff we incur from trying to analyze the request before async-graphql does it?